object_storage_api/repositories/image.py

```diff
@@ -56,3 +56,27 @@ def get(self, image_id: str, session: ClientSession = None) -> Optional[ImageOut
         if image:
             return ImageOut(**image)
         return None
+
+    def list(self, entity_id: Optional[str], primary: Optional[bool], session: ClientSession = None) -> list[ImageOut]:
+        """
+        Retrieve Images from a MongoDB database.
+
+        :param session: PyMongo ClientSession to use for database operations.
+        :return: List of Images or an empty list if no Images are retrieved
+        """
+
+        query = {}
+        if entity_id is not None:
+            query["entity_id"] = CustomObjectId(entity_id)
+        if primary is not None:
+            query["primary"] = primary
+
+        message = "Retrieving all images from the database"
+        if not query:
+            logger.info(message)
+        else:
+            logger.info("%s matching the provided filter(s)", message)
+            logger.debug("Provided filter(s): %s", query)
+
+        images = self._images_collection.find(query, session=session)
+        return [ImageOut(**image) for image in images]
```

Comment:
These are already ImageOut's as they come from the database, although contrary to IMS I would prefer them converted to schemas in the service layer rather than the router layer so I would suggest using this and removing the conversion in the `router.py`.
```suggestion
        return [ImageSchema(**image) for image in images]
```

Reply:
Just double checking, if I am converting them to schemas in the service layer, would I not be returning as them `ImageOut` in the Repo layer like so

`return [ image for image in images]`
instead of 
`return [ImageSchema(**image) for image in images]`